extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_decoder.c

```diff
@@ -1025,6 +1029,16 @@ ArrowErrorCode ArrowIpcDecoderDecodeHeader(struct ArrowIpcDecoder* decoder,
   NANOARROW_RETURN_NOT_OK(
       ArrowIpcDecoderCheckHeader(decoder, &data, &decoder->header_size_bytes, error));
 
+  // Check that data contains at least the entire header (return ESPIPE to signal
+  // that reading more data may help).
+  int64_t message_body_size = decoder->header_size_bytes - 8;
```

Comment:
nit: above we use `2 * sizeof(int32_t)` but here we hardcode 8; maybe just pull it out into a constant?